sdk/storage/Azure.Storage.DataMovement/src/Models/CopyMethod.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.Storage.DataMovement.Models
+{
+    /// <summary>
+    /// Copy Method.
+    /// Users can specify what type copy to occur.
+    /// </summary>
+    public enum CopyMethod
+    {
+        /// <summary>
+        /// Dowloads data form source to memory, then uploads the data from memory to destination
```

Comment:
I wonder if we should offer disk buffering as well for cases where memory isn't enough.

Reply:
I added an MaximumBufferSize in the BlobSyncCopyOptions